### PR TITLE
Add Ruby config file option to installer

### PR DESCRIPTION
**Pull request by tombruijn**
Based on #1324

## Rename matchers in installer spec

Differentiate between YAML install method and others in the matcher names so it's easier to add ones for the Ruby config file later.

## Add Ruby config file option to installer

To replace the YAML config file people need to know they can configure the gem with a Ruby config file too. Add this option to the installer so people can choose this config option when installing AppSignal in their application.

I've made it the first listed method as it's the most flexible configuration method. The YAML file option has the "legacy" label next to it now to discourage people from using it.

### CODE REVIEW

**Review by luismiramirez — APPROVED** [review]

File: resources/appsignal.rb.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  #   "ScriptError",
+  #   "LoadError",
+  #   "NotImplementedError",
+  #   "SyntaxError",
+  #   "SecurityError",
+  #   "SignalException",
+  #   "Interrupt",
+  #   "SystemExit",
+  #   "SystemStackError"
+  # ]
```
Comment (unflxw) [inline resources/appsignal.rb.erb:31]: Do we want to suggest ignoring all of these out of the box? I wonder if people will uncomment this without thinking and accidentally miss on errors. Perhaps suggest something like:

```suggestion
  # config.ignore_errors += [
  #   "MyCustomError",
  # ]
```

**Review by unflxw — APPROVED** [review]
> The YAML file option has the "legacy" label next to it now to discourage people from using it.

Could/should we remove it entirely? 🤔 The installer is mostly used by new AppSignal users, I don't think they'll miss it.

Either way, now that the YAML is considered legacy, are there things that need to be changed regarding how we document configuration options? Perhaps the order of the code snippets and sections [in the configuration overview page](https://docs.appsignal.com/ruby/configuration.html) should be changed to put `.rb` first.

File: resources/appsignal.rb.erb (hunk 1 above)
Comment (tombruijn) [reply resources/appsignal.rb.erb:31]: Sure. I just copied the template from the appsignal.yml file 🤷‍♂️

**Comment by tombruijn** [comment]
> > The YAML file option has the "legacy" label next to it now to discourage people from using it.
> 
> Could/should we remove it entirely? 🤔 The installer is mostly used by new AppSignal users, I don't think they'll miss it.

I don't want to shock and confuse existing customers that are used to configure with the YAML file and don't have the time to adapt to the new thing right away. So I don't want to introduce a new thing and take out the old way in the same version. It's legacy now and we can choose to remove it in a version before the next major version.

> Either way, now that the YAML is considered legacy, are there things that need to be changed regarding how we document configuration options? Perhaps the order of the code snippets and sections [in the configuration overview page](https://docs.appsignal.com/ruby/configuration.html) should be changed to put `.rb` first.

Yes, we should update the docs to make the `appsignal.rb` file method first class.

**Pull request merged**